src/pint/models/parameter.py

```diff
@@ -1418,6 +1418,24 @@ def __repr__(self):
 
         return out
 
+    def create_new_index_copy(self, index):
+        """Sets up copy of current param with a new index."""
```

Comment:
How about adding a bit more to this docstring about how this method is intended to be used?